src/main/java/net/floodlightcontroller/websocket/WebSocketManager.java

```diff
@@ -0,0 +1,257 @@
+package net.floodlightcontroller.websocket;
+
+import java.io.IOException;  
+import java.util.Collection;    
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jetty.server.Server; 
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.websocket.api.Session;
+
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import net.floodlightcontroller.core.module.FloodlightModuleException;
+import net.floodlightcontroller.core.module.IFloodlightModule;
+import net.floodlightcontroller.core.module.IFloodlightService;
+
+import net.floodlightcontroller.restserver.IRestApiService;
+import net.floodlightcontroller.storage.IResultSet;
+import net.floodlightcontroller.storage.IStorageSourceListener;
+import net.floodlightcontroller.storage.IStorageSourceService;
+import net.floodlightcontroller.storage.StorageException;
+
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+
+
+/**
+ * Module starts a Jetty server, and registers storage listeners for real-time table changes
+ *
+ * TODO: Check to make sure all threads are finished correctly.
```

Comment:
Is this still relevant?

Reply:
I dont think I have any threading problems, but it would be good to check to make sure it is completely optimised.